packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -260,42 +283,21 @@ export class UnreferencedStateTracker {
         // Also, start a timer for the sweep timeout.
         if (unreferencedDurationMs >= this.inactiveTimeoutMs) {
             this._state = UnreferencedState.Inactive;
-            this.clearTimers();
+            this.inactiveTimer.clear();
 
             if (this.sweepTimeoutMs !== undefined) {
-                setLongTimeout(
-                    this.sweepTimeoutMs - unreferencedDurationMs,
-                    () => { this._state = UnreferencedState.SweepReady; },
-                    (timer) => { this.sweepTimer = timer; },
-                );
+                this.sweepTimer.restart(this.sweepTimeoutMs - unreferencedDurationMs);
```

Comment:
Shouldn't we be starting the sweep timer here?

Reply:
restart just calls start if it's not already started

Reply:
Nit: it seems like there is duplicate logic in the constructor

Reply:
Took a second look here - This one needs to be restart for sure. Every time we summarize past the `inactiveTimeoutMs` we'll come through here again and again.

And the code in the constructor is different in two ways:
* Doesn't need to call `this.inactiveTimer.clear()`
* Uses a different timeout value